packages/plugins/plugin-builder-flowtype/src/transformer/index.ts

```diff
@@ -162,29 +167,63 @@ export function createTransformer(importHelpers: boolean) {
                           return t.emptyStatement();
                         }
                       }),
-                    ),
+                      // { [key: type]: value }
+                      ...(node.indexers
+                        ? node.indexers.map((indexer) => {
+                            return t.expressionStatement(
+                              t.callExpression(
+                                t.memberExpression(
+                                  t.identifier('typeLiteral'),
+                                  t.identifier('computedProperty'),
+                                ),
+                                [
+                                  t.stringLiteral(
+                                    indexer.id ? indexer.id.name : 'key',
+                                  ),
+                                  transformLiteral(indexer.key),
+                                  transformLiteral(indexer.value),
+                                  t.numericLiteral(0), // ComputedPropertyOperator.INDEX_SIGNATURE
```

Comment:
Maybe we should pause a little bit on indexers. The current runtime support for indexer is not very correct.